polygeist/tools/cgeist/Lib/ValueCategory.cc

```diff
@@ -380,3 +393,35 @@ ValueCategory ValueCategory::IntCast(OpBuilder &Builder, Type PromotionType,
 
   return {Res, /*IsReference*/ false};
 }
+
+ValueCategory ValueCategory::ICmpNE(mlir::OpBuilder &builder,
+                                    ValueCategory RHS) const {
+  return ICmp(builder, arith::CmpIPredicate::ne, RHS);
+}
+
+ValueCategory ValueCategory::FCmpUNE(mlir::OpBuilder &builder,
+                                     ValueCategory RHS) const {
+  return FCmp(builder, arith::CmpFPredicate::UNE, RHS);
+}
+
+ValueCategory ValueCategory::ICmp(OpBuilder &builder,
+                                  arith::CmpIPredicate predicate,
+                                  ValueCategory RHS) const {
```

Comment:
why not takes RHS as a Value, as only `RHS.val` is used?

Reply:
Just for symmetry. There is a bunch of functions also receiving `ValueCategory` instead of `Value`. I'll create a PR on top of this one to change the type passed here if that works for you.